src/server/pmix_server.c

```diff
@@ -294,17 +298,39 @@ PMIX_EXPORT pmix_status_t PMIx_server_init(pmix_server_module_t *module,
                     lt->owner = info[n].value.data.uint32;
                     lt->owner_given = true;
                 }
-            } else if (0 == strcmp(info[n].key, PMIX_GRPID)) {
+            } 
+	    else if (0 == strcmp(info[n].key, PMIX_GRPID)) {
                /* the grpid is in the uint32_t storage */
                 PMIX_LIST_FOREACH(lt, &pmix_server_globals.listeners, pmix_listener_t) {
                     lt->group = info[n].value.data.uint32;
                     lt->group_given = true;
-                }
+		}
             } else if (0 == strcmp(info[n].key, PMIX_SOCKET_MODE)) {
                 /* socket mode is in the uint32_t storage */
                 PMIX_LIST_FOREACH(lt, &pmix_server_globals.listeners, pmix_listener_t) {
                     lt->mode = info[n].value.data.uint32;
-                }
+                } 
+	    } else if (0 == strcmp(info[n].key, PMIX_SERVER_TOOL_SUPPORT)) {
+                 /* This is where we need to setup a listening
+                 * address and store it in the /tmp/pmix.nodename.tool
+                 * file, where nodename = the local hostname. We will
+                 * then need to call pmix_start_listening(&thataddress)
+                 */ 
+		
+                pmix_listener_t *tl = PMIX_NEW(pmix_listener_t);
+                memset(tl, 0, sizeof(pmix_listener_t));
+		
+		memset(&tl->address, 0, sizeof(struct sockaddr_un));
+		tl -> address.sun_family = AF_UNIX;
+		tl -> mode = ACCESSPERMS;
+
+		/* Get up to 10 chars of hostname.*/
+		gethostname(myhostname, myhostnamelen);
+                snprintf(tl->address.sun_path, sizeof(tl->address.sun_path) - 1, "/tmp/pmix.%s.%d", myhostname, getpid());
```

Comment:
Ah, I see - you made the file be an actual Unix domain socket itself, as opposed to containing info. Okay.